### ISSUE TITLE
Tablet health check occasionally does not identify new tablet type after failover

### ISSUE THREAD

**Issue by aquarapid** [vitessio/vitess#6366]
This has been seen in more recent releases as well (specifically 7f1dae86ea91aada6d4e28b39698affac64a374b).  Wanted to make sure we understand why this was/is happening and if we can make sure the tablet type is updated successfully after a failover.

See discussion here:  https://vitess.slack.com/archives/CMKTCUYNQ/p1591511700036100

Some of the relevant logs:

```
I0607 17:30:46.469597       1 controller.go:189] trying to find a tablet eligible for vreplication. stream id: 8
I0607 17:30:46.469668       1 controller.go:194] found a tablet eligible for vreplication. stream id: 8  tablet: cell:"uscentral1" uid:699290100
E0607 15:39:52.114559       1 controller.go:149] stream 8: Code: FAILED_PRECONDITION
vttablet: rpc error: code = FailedPrecondition desc = invalid tablet type: REPLICA, want: MASTER or []
, retrying after 5s
I0607 15:39:52.910788       1 healthcheck.go:470] HealthCheckUpdate(Type Change): uscentral1-0699290100, tablet: uscentral1-699290100 (uscentral1-workspaces-0-replica-a-0.vttablet), target workspaces/0 (REPLICA) => workspaces/0 (MASTER), reparent time: 1591544392
I0607 15:39:53.102758       1 healthcheck.go:470] HealthCheckUpdate(Type Change): uscentral1-0170756700, tablet: uscentral1-170756700 (uscentral1-workspaces-0-replica-b-0.vttablet), target workspaces/0 (MASTER) => workspaces/0 (REPLICA), reparent time: 1591251921
I0607 15:39:57.115590       1 controller.go:189] trying to find a tablet eligible for vreplication. stream id: 8
I0607 15:39:57.115656       1 controller.go:194] found a tablet eligible for vreplication. stream id: 8  tablet: cell:"uscentral1" uid:699290100
```

However, uscentral1-699290100 was in fact a master (see discussion in Slack).

**Comment by deepthi** [comment]
This should be fixed by #6442, needs to be re-tested.

**Comment by GuptaManan100** [comment]
Closing this issue. Reopen if the error persists
